sql/core/src/main/scala/org/apache/spark/sql/sources/interfaces.scala

```diff
@@ -544,11 +544,35 @@ abstract class HadoopFsRelation private[sql](maybePartitionSpec: Option[Partitio
   }
 
   private def discoverPartitions(): PartitionSpec = {
-    val typeInference = sqlContext.conf.partitionColumnTypeInferenceEnabled()
     // We use leaf dirs containing data files to discover the schema.
     val leafDirs = fileStatusCache.leafDirToChildrenFiles.keys.toSeq
-    PartitioningUtils.parsePartitions(leafDirs, PartitioningUtils.DEFAULT_PARTITION_NAME,
-      typeInference)
+    userDefinedPartitionColumns match {
+      case Some(schema) =>
+        val spec = PartitioningUtils.parsePartitions(
+          leafDirs, PartitioningUtils.DEFAULT_PARTITION_NAME, false)
```

Comment:
Super-minor nit: could you explicitly name the boolean parameter here at the call-site, e.g. `inferSchema = false`? This is one of IntelliJ's automatic style recommendations and I'm a fan of it because it makes the code a bit easier to read. I might also just change this myself on merge.